src/looprestoration.rs

```diff
@@ -58,93 +59,127 @@ pub struct Dav1dLoopRestorationDSPContext {
-                pub(crate) fn $name(
-                    dst: *mut pixel,
-                    dst_stride: ptrdiff_t,
-                    left: const_left_pixel_row,
-                    lpf: *const pixel,
-                    w: libc::c_int,
-                    h: libc::c_int,
-                    params: *const LooprestorationParams,
-                    edges: LrEdgeFlags,
-                    bitdepth_max: libc::c_int,
-                );
-            )*
+/// Generates a generic wrapper function that delegates to the appropriate
+/// bitdepth-specific extern function.
+macro_rules! decl_looprestorationfilter_bd_fn {
+    ($name:ident, $suffix:ident) => {
+        paste::paste! {
+            #[cfg(feature = "asm")]
+            unsafe extern "C" fn [<$name $suffix>]<BD: BitDepth>(
+                dst: *mut pixel,
+                dst_stride: ptrdiff_t,
+                left: const_left_pixel_row,
+                lpf: *const pixel,
+                w: libc::c_int,
+                h: libc::c_int,
+                params: *const LooprestorationParams,
+                edges: LrEdgeFlags,
+                bitdepth_max: libc::c_int,
+            ) {
+                (match BD::BPC {
+                    BPC::BPC8 => {
+                        extern "C" {
+                            fn [<$name _8bpc $suffix>](
+                                dst: *mut pixel,
+                                dst_stride: ptrdiff_t,
+                                left: const_left_pixel_row,
+                                lpf: *const pixel,
+                                w: libc::c_int,
+                                h: libc::c_int,
+                                params: *const LooprestorationParams,
+                                edges: LrEdgeFlags,
+                                bitdepth_max: libc::c_int,
+                            );
+                        }
+                        [<$name _8bpc $suffix>]
+                    }
+
+                    BPC::BPC16 => {
+                        extern "C" {
+                            fn [<$name _16bpc $suffix>](
+                                dst: *mut pixel,
+                                dst_stride: ptrdiff_t,
+                                left: const_left_pixel_row,
+                                lpf: *const pixel,
+                                w: libc::c_int,
+                                h: libc::c_int,
+                                params: *const LooprestorationParams,
+                                edges: LrEdgeFlags,
+                                bitdepth_max: libc::c_int,
+                            );
+                        }
+                        [<$name _16bpc $suffix>]
+                    }
+                })(dst, dst_stride, left, lpf, w, h, params, edges, bitdepth_max)
+            }
         }
     };
 }
 
+decl_looprestorationfilter_bd_fn!(dav1d_wiener_filter7, _ssse3);
```

Comment:
If we don't globally declare these but leave them as local expressions in the init function, then we won't need the wrapper functions.